src/java_bytecode/java_class_loader.h

```diff
@@ -28,6 +28,16 @@ class java_class_loadert:public messaget
   void set_java_cp_include_files(std::string &);
   void add_load_classes(const std::vector<irep_idt> &);
 
+  /// A function that yields a list of extra dependencies based on a class name.
+  typedef std::function<std::vector<irep_idt>(const irep_idt &)>
+    get_extra_class_refs_functiont;
+
+  /// Sets a function that provides extra dependencies for a particular class.
```

Comment:
While I think this is a better place for documentation - I think we currently put all docs with the implementation so I guess we should stick with that?